src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileNamesExtractor.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import com.google.common.base.Strings;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Component
+@ConditionalOnProperty(value = "scheduling.task.scan.enabled", matchIfMissing = true)
+public class FileNamesExtractor {
+    private static final Logger log = LoggerFactory.getLogger(FileNamesExtractor.class);
+
+    private FileNamesExtractor() {
+        // Utility class
+    }
+
+    public static List<String> getZipFileNamesFromContainer(CloudBlobContainer container) {
```

Comment:
since it's a utility, perhaps something like `shuffleContainerBlobs` will suit? mind it's a list of strings which signifies file names so something along those words would be ideal :upside_down_face: